### ISSUE TITLE
Types not checked in finally block after try block returns

### ISSUE THREAD

**Issue by joe-pierce** [astral-sh/ty#3003]
### Summary

I may be missing something obvious here but Ty doesn't seem to be catching type errors from within a finally clause (of a try/finally) block if there's a return in the try. In the playground it suggests that Ty thinks the finally is never reached, but it should be executed still.

**ty version:** 0.0.21
**Python:** 3.14.1

## Minimal Reproduction

Can be found in playground [here](https://play.ty.dev/09a6cdc4-2850-4144-bb6f-74f194c75a38)

```python
import asyncio
from dataclasses import dataclass

@dataclass
class Item:
    pass

class ItemQueue(asyncio.Queue[Item]):
    """Typed queue that only accepts ``Item`` objects."""

queue = ItemQueue()
def wrapper(*args, **kwargs) -> str:
    try:
        return "test"
    finally:
        queue.put_nowait("bad_object_type") <-- Doesn't get caught by Ty
```
The type error is correctly caught when that `queue.put_nowait("bad_object_type")` line is put before the return. 

### Version

0.0.21

**Comment by AlexWaygood** [comment]
Thanks for the report. I think this is covered by #233
